File: smallpict-java/src/main/java/com/smallpict/models/JobStatusResult.java
package com.smallpict.models;

import com.fasterxml.jackson.annotation.JsonCreator;
import com.fasterxml.jackson.annotation.JsonProperty;
import java.util.Map;

public class JobStatusResult {
    @JsonProperty("job_id")
    private final String jobId;
    private final String status;
    private final String url;
    private final String format;
    @JsonProperty("bytes_saved")
    private final Long bytesSaved;
    private final Map<String, Object> error;
    @JsonProperty("created_at")
    private final String createdAt;
    @JsonProperty("updated_at")
    private final String updatedAt;

    @JsonCreator
    public JobStatusResult(
            @JsonProperty("job_id") String jobId,
            @JsonProperty("status") String status,
            @JsonProperty("url") String url,
            @JsonProperty("format") String format,
            @JsonProperty("bytes_saved") Long bytesSaved,
            @JsonProperty("error") Map<String, Object> error,
            @JsonProperty("created_at") String createdAt,
            @JsonProperty("updated_at") String updatedAt) {
        this.jobId = jobId;
        this.status = status != null ? status : "processing";
        this.url = url;
        this.format = format;
        this.bytesSaved = bytesSaved;
        this.error = error;
        this.createdAt = createdAt;
        this.updatedAt = updatedAt;
    }

    public String getJobId() {
        return jobId;
    }

    public String getStatus() {
        return status;
    }

    public String getUrl() {
        return url;
    }

    public String getFormat() {
        return format;
    }

    public Long getBytesSaved() {
        return bytesSaved;
    }

    public Map<String, Object> getError() {
        return error;
    }

    public String getCreatedAt() {
        return createdAt;
    }

    public String getUpdatedAt() {
        return updatedAt;
    }
}

File: smallpict-php/src/Models/JobStatusResult.php
<?php

declare(strict_types=1);

namespace SmallPict\Models;

class JobStatusResult
{
    private string $jobId;
    private string $status;
    private ?string $url;
    private ?string $format;
    private ?int $bytesSaved;
    /**
     * @var array<string, mixed>|null
     */
    private ?array $error;
    private ?string $createdAt;
    private ?string $updatedAt;

    /**
     * @param string $jobId
     * @param string $status
     * @param string|null $url
     * @param string|null $format
     * @param int|null $bytesSaved
     * @param array<string, mixed>|null $error
     * @param string|null $createdAt
     * @param string|null $updatedAt
     */
    public function __construct(
        string $jobId,
        string $status,
        ?string $url = null,
        ?string $format = null,
        ?int $bytesSaved = null,
        ?array $error = null,
        ?string $createdAt = null,
        ?string $updatedAt = null
    ) {
        $this->jobId = $jobId;
        $this->status = $status;
        $this->url = $url;
        $this->format = $format;
        $this->bytesSaved = $bytesSaved;
        $this->error = $error;
        $this->createdAt = $createdAt;
        $this->updatedAt = $updatedAt;
    }

    public function getJobId(): string
    {
        return $this->jobId;
    }

    public function getStatus(): string
    {
        return $this->status;
    }

    public function getUrl(): ?string
    {
        return $this->url;
    }

    public function getFormat(): ?string
    {
        return $this->format;
    }

    public function getBytesSaved(): ?int
    {
        return $this->bytesSaved;
    }

    /**
     * @return array<string, mixed>|null
     */
    public function getError(): ?array
    {
        return $this->error;
    }

    public function getCreatedAt(): ?string
    {
        return $this->createdAt;
    }

    public function getUpdatedAt(): ?string
    {
        return $this->updatedAt;
    }
}
